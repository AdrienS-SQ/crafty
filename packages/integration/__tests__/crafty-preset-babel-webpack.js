/* global describe, it, expect */

const fs = require("fs");
const path = require("path");

const rimraf = require("rimraf");

const testUtils = require("../utils");

it("Compiles JavaScript", () => {
  process.chdir(
    path.join(__dirname, "../fixtures/crafty-preset-babel-webpack/compiles")
  );
  rimraf.sync("dist");

  const result = testUtils.run(["run", "default"]);

  expect(result).toMatchSnapshot();

  expect(fs.existsSync("dist/js/myBundle.min.js")).toBeTruthy();
  expect(fs.existsSync("dist/js/myBundle.min.js.map")).toBeTruthy();

  expect(
    fs.readFileSync("dist/js/myBundle.min.js").toString("utf8")
  ).toMatchSnapshot();
});

it("Lints JavaScript with webpack", () => {
  process.chdir(
    path.join(__dirname, "../fixtures/crafty-preset-babel-webpack/lints")
  );
  rimraf.sync("dist");

  const result = testUtils.run(["run", "default"]);

  expect(result).toMatchSnapshot();

  // Files aren't generated on failed lint
  expect(fs.existsSync("dist/js/myBundle.min.js")).toBeFalsy();
  expect(fs.existsSync("dist/js/myBundle.min.js.map")).toBeFalsy();
});

it("Fails gracefully on broken markup", () => {
  process.chdir(
    path.join(__dirname, "../fixtures/crafty-preset-babel-webpack/fails")
  );
  rimraf.sync("dist");

  const result = testUtils.run(["run", "default"]);

  expect(result).toMatchSnapshot();

  // Files aren't generated on failed lint
  expect(fs.existsSync("dist/js/myBundle.min.js")).toBeFalsy();
  expect(fs.existsSync("dist/js/myBundle.min.js.map")).toBeFalsy();
});

it("Removes unused classes", () => {
  process.chdir(
    path.join(__dirname, "../fixtures/crafty-preset-babel-webpack/tree-shaking")
  );
  rimraf.sync("dist");

  const result = testUtils.run(["run", "default"]);

  expect(result).toMatchSnapshot();

  expect(fs.existsSync("dist/js/myBundle.min.js")).toBeTruthy();
  expect(fs.existsSync("dist/js/myBundle.min.js.map")).toBeTruthy();

  const content = fs.readFileSync("dist/js/myBundle.min.js").toString("utf8");

  expect(content.indexOf("From class A") > -1).toBeTruthy();
  expect(content.indexOf("From class C") > -1).toBeTruthy();
  expect(content.indexOf("From class D") > -1).toBeFalsy();
  expect(content.indexOf("From class B") > -1).toBeFalsy();
});
