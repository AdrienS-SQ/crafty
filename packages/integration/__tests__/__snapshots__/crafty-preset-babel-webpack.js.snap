// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Compiles JavaScript 1`] = `
Object {
  "status": 0,
  "stderr": "",
  "stdout": "
[__:__:__] Starting Crafty __version__...
[__:__:__] Files will be stored in:
            js: __PATH__/packages/integration/fixtures/crafty-preset-babel-webpack/compiles/dist/js
[__:__:__] Starting 'default' ...
[__:__:__] Starting 'js' ...
[__:__:__] Starting 'js_myBundle' ...
  size  name     module                      status
  1 kB  0        ./js/script.js + 1 modules  built
  1 kB  1        multi ./js/script.js        built

  size  name     asset                       status
  1 kB  default  myBundle.min.js             emitted
  1 kB  default  myBundle.min.js.map         emitted



  Compiled successfully!
  Δt ____ms

[__:__:__] Finished 'js_myBundle' after ____ ms
[__:__:__] Finished 'js' after ____ ms
[__:__:__] Finished 'default' after ____ ms
",
}
`;

exports[`Compiles JavaScript 2`] = `
"!function(e,t){if(\\"object\\"==typeof exports&&\\"object\\"==typeof module)module.exports=t();else if(\\"function\\"==typeof define&&define.amd)define([],t);else{var n=t();for(var r in n)(\\"object\\"==typeof exports?exports:e)[r]=n[r]}}(window,function(){return function(e){var t={};function n(r){if(t[r])return t[r].exports;var o=t[r]={i:r,l:!1,exports:{}};return e[r].call(o.exports,o,o.exports,n),o.l=!0,o.exports}return n.m=e,n.c=t,n.d=function(e,t,r){n.o(e,t)||Object.defineProperty(e,t,{configurable:!1,enumerable:!0,get:r})},n.r=function(e){Object.defineProperty(e,\\"__esModule\\",{value:!0})},n.n=function(e){var t=e&&e.__esModule?function(){return e.default}:function(){return e};return n.d(t,\\"a\\",t),t},n.o=function(e,t){return Object.prototype.hasOwnProperty.call(e,t)},n.p=\\"\\",n(n.s=1)}([function(e,t,n){\\"use strict\\";n.r(t);var r=function(){function e(e,t){for(var n=0;n<t.length;n++){var r=t[n];r.enumerable=r.enumerable||!1,r.configurable=!0,\\"value\\"in r&&(r.writable=!0),Object.defineProperty(e,r.key,r)}}return function(t,n,r){return n&&e(t.prototype,n),r&&e(t,r),t}}();function o(e,t){if(!e)throw new ReferenceError(\\"this hasn't been initialised - super() hasn't been called\\");return!t||\\"object\\"!=typeof t&&\\"function\\"!=typeof t?e:t}var i=function(e){function t(){var e,n,r;!function(e,t){if(!(e instanceof t))throw new TypeError(\\"Cannot call a class as a function\\")}(this,t);for(var i=arguments.length,u=Array(i),c=0;c<i;c++)u[c]=arguments[c];return n=r=o(this,(e=t.__proto__||Object.getPrototypeOf(t)).call.apply(e,[this].concat(u))),r.handleClick=function(){r.setState({hey:!0})},o(r,n)}return function(e,t){if(\\"function\\"!=typeof t&&null!==t)throw new TypeError(\\"Super expression must either be null or a function, not \\"+typeof t);e.prototype=Object.create(t&&t.prototype,{constructor:{value:e,enumerable:!1,writable:!0,configurable:!0}}),t&&(Object.setPrototypeOf?Object.setPrototypeOf(e,t):e.__proto__=t)}(t,React.Component),r(t,[{key:\\"render\\",value:function(){return React.createElement(\\"div\\",{onClick:this.handleClick},\\"Hey \\",this.state.hey)}}]),t}(),u=function(){function e(e,t){for(var n=0;n<t.length;n++){var r=t[n];r.enumerable=r.enumerable||!1,r.configurable=!0,\\"value\\"in r&&(r.writable=!0),Object.defineProperty(e,r.key,r)}}return function(t,n,r){return n&&e(t.prototype,n),r&&e(t,r),t}}();var c=function(){function e(){!function(e,t){if(!(e instanceof t))throw new TypeError(\\"Cannot call a class as a function\\")}(this,e),console.log(\\"init class\\")}return u(e,[{key:\\"method\\",value:function(){console.log(i)}}]),e}();t.default=c},function(e,t,n){e.exports=n(0)}])});
//# sourceMappingURL=myBundle.min.js.map"
`;

exports[`Compiles JavaScript with externals 1`] = `
Object {
  "status": 0,
  "stderr": "",
  "stdout": "
[__:__:__] Starting Crafty __version__...
[__:__:__] Files will be stored in:
            js: __PATH__/packages/integration/fixtures/crafty-preset-babel-webpack/externals/dist/js
[__:__:__] Starting 'default' ...
[__:__:__] Starting 'js' ...
[__:__:__] Starting 'js_myBundle' ...
  size  name     module                              status
  1 kB  0        external \\"somelibrary/thing/stuff\\"  built
  1 kB  1        ./js/script.js + 1 modules          built
  1 kB  2        multi ./js/script.js                built

  size  name     asset                               status
  1 kB  default  myBundle.min.js                     emitted
  1 kB  default  myBundle.min.js.map                 emitted



  Compiled successfully!
  Δt ____ms

[__:__:__] Finished 'js_myBundle' after ____ ms
[__:__:__] Finished 'js' after ____ ms
[__:__:__] Finished 'default' after ____ ms
",
}
`;

exports[`Compiles JavaScript with externals 2`] = `
"!function(e,t){if(\\"object\\"==typeof exports&&\\"object\\"==typeof module)module.exports=t(require(\\"somelibrary/thing/stuff\\"));else if(\\"function\\"==typeof define&&define.amd)define([\\"somelibrary/thing/stuff\\"],t);else{var n=\\"object\\"==typeof exports?t(require(\\"somelibrary/thing/stuff\\")):t(e[\\"somelibrary/thing/stuff\\"]);for(var r in n)(\\"object\\"==typeof exports?exports:e)[r]=n[r]}}(window,function(e){return function(e){var t={};function n(r){if(t[r])return t[r].exports;var o=t[r]={i:r,l:!1,exports:{}};return e[r].call(o.exports,o,o.exports,n),o.l=!0,o.exports}return n.m=e,n.c=t,n.d=function(e,t,r){n.o(e,t)||Object.defineProperty(e,t,{configurable:!1,enumerable:!0,get:r})},n.r=function(e){Object.defineProperty(e,\\"__esModule\\",{value:!0})},n.n=function(e){var t=e&&e.__esModule?function(){return e.default}:function(){return e};return n.d(t,\\"a\\",t),t},n.o=function(e,t){return Object.prototype.hasOwnProperty.call(e,t)},n.p=\\"\\",n(n.s=2)}([function(t,n){t.exports=e},function(e,t,n){\\"use strict\\";n.r(t);var r=n(0),o=n.n(r),i=function(){function e(e,t){for(var n=0;n<t.length;n++){var r=t[n];r.enumerable=r.enumerable||!1,r.configurable=!0,\\"value\\"in r&&(r.writable=!0),Object.defineProperty(e,r.key,r)}}return function(t,n,r){return n&&e(t.prototype,n),r&&e(t,r),t}}();function u(e,t){if(!e)throw new ReferenceError(\\"this hasn't been initialised - super() hasn't been called\\");return!t||\\"object\\"!=typeof t&&\\"function\\"!=typeof t?e:t}var f=function(e){function t(){var e,n,r;!function(e,t){if(!(e instanceof t))throw new TypeError(\\"Cannot call a class as a function\\")}(this,t);for(var o=arguments.length,i=Array(o),f=0;f<o;f++)i[f]=arguments[f];return n=r=u(this,(e=t.__proto__||Object.getPrototypeOf(t)).call.apply(e,[this].concat(i))),r.handleClick=function(){r.setState({hey:!0})},u(r,n)}return function(e,t){if(\\"function\\"!=typeof t&&null!==t)throw new TypeError(\\"Super expression must either be null or a function, not \\"+typeof t);e.prototype=Object.create(t&&t.prototype,{constructor:{value:e,enumerable:!1,writable:!0,configurable:!0}}),t&&(Object.setPrototypeOf?Object.setPrototypeOf(e,t):e.__proto__=t)}(t,React.Component),i(t,[{key:\\"render\\",value:function(){return React.createElement(\\"div\\",{onClick:this.handleClick},\\"Hey \\",this.state.hey)}}]),t}(),c=function(){function e(e,t){for(var n=0;n<t.length;n++){var r=t[n];r.enumerable=r.enumerable||!1,r.configurable=!0,\\"value\\"in r&&(r.writable=!0),Object.defineProperty(e,r.key,r)}}return function(t,n,r){return n&&e(t.prototype,n),r&&e(t,r),t}}();var a=function(){function e(){!function(e,t){if(!(e instanceof t))throw new TypeError(\\"Cannot call a class as a function\\")}(this,e),console.log(\\"init class\\")}return c(e,[{key:\\"method\\",value:function(){console.log(f,o.a)}}]),e}();t.default=a},function(e,t,n){e.exports=n(1)}])});
//# sourceMappingURL=myBundle.min.js.map"
`;

exports[`Does not transpile on modern browsers 1`] = `
Object {
  "status": 0,
  "stderr": "",
  "stdout": "
[__:__:__] Starting Crafty __version__...
[__:__:__] Files will be stored in:
            js: __PATH__/packages/integration/fixtures/crafty-preset-babel-webpack/no-old-browser/dist/js
[__:__:__] Starting 'default' ...
[__:__:__] Starting 'js' ...
[__:__:__] Starting 'js_myBundle' ...
  size  name     module                      status
  1 kB  0        ./js/script.js + 1 modules  built
  1 kB  1        multi ./js/script.js        built

  size  name     asset                       status
  1 kB  default  myBundle.min.js             emitted
  1 kB  default  myBundle.min.js.map         emitted



  Compiled successfully!
  Δt ____ms

[__:__:__] Finished 'js_myBundle' after ____ ms
[__:__:__] Finished 'js' after ____ ms
[__:__:__] Finished 'default' after ____ ms
",
}
`;

exports[`Does not transpile on modern browsers 2`] = `
"!function(e,t){if(\\"object\\"==typeof exports&&\\"object\\"==typeof module)module.exports=t();else if(\\"function\\"==typeof define&&define.amd)define([],t);else{var n=t();for(var o in n)(\\"object\\"==typeof exports?exports:e)[o]=n[o]}}(window,function(){return function(e){var t={};function n(o){if(t[o])return t[o].exports;var r=t[o]={i:o,l:!1,exports:{}};return e[o].call(r.exports,r,r.exports,n),r.l=!0,r.exports}return n.m=e,n.c=t,n.d=function(e,t,o){n.o(e,t)||Object.defineProperty(e,t,{configurable:!1,enumerable:!0,get:o})},n.r=function(e){Object.defineProperty(e,\\"__esModule\\",{value:!0})},n.n=function(e){var t=e&&e.__esModule?function(){return e.default}:function(){return e};return n.d(t,\\"a\\",t),t},n.o=function(e,t){return Object.prototype.hasOwnProperty.call(e,t)},n.p=\\"\\",n(n.s=1)}([function(e,t,n){\\"use strict\\";n.r(t);class o extends React.Component{constructor(...e){var t;return t=super(...e),this.handleClick=(()=>{this.setState({hey:!0})}),t}render(){return React.createElement(\\"div\\",{onClick:this.handleClick},\\"Hey \\",this.state.hey)}}n.d(t,\\"default\\",function(){return r});class r{constructor(){console.log(\\"init class\\")}method(){console.log(o)}}},function(e,t,n){e.exports=n(0)}])});
//# sourceMappingURL=myBundle.min.js.map"
`;

exports[`Fails gracefully on broken markup 1`] = `
Object {
  "status": 1,
  "stderr": "[js_myBundle] (!) Errors were found
js/script.js
[__:__:__] 'js_myBundle' errored after ____ ms
[__:__:__] Error: Errors were found
    ...stacktrace...
[__:__:__] 'js' errored after ____ ms
[__:__:__] 'default' errored after ____ ms
",
  "stdout": "
[__:__:__] Starting Crafty __version__...
[__:__:__] Files will be stored in:
            js: __PATH__/packages/integration/fixtures/crafty-preset-babel-webpack/fails/dist/js
[__:__:__] Starting 'default' ...
[__:__:__] Starting 'js' ...
[__:__:__] Starting 'js_myBundle' ...

__PATH__/packages/integration/fixtures/crafty-preset-babel-webpack/fails/js/script.js
  15:1  error  Parsing error: Unexpected token

  13 |
  14 | }
> 15 |
     | ^

✖ 1 problem (1 error, 0 warnings)

",
}
`;

exports[`Lints JavaScript with webpack 1`] = `
Object {
  "status": 1,
  "stderr": "[__:__:__] 'js_myBundle' errored after ____ ms
[__:__:__] Error: Webpack compilation failed
    ...stacktrace...
[__:__:__] 'js' errored after ____ ms
[__:__:__] 'default' errored after ____ ms
",
  "stdout": "
[__:__:__] Starting Crafty __version__...
[__:__:__] Files will be stored in:
            js: __PATH__/packages/integration/fixtures/crafty-preset-babel-webpack/lints/dist/js
[__:__:__] Starting 'default' ...
[__:__:__] Starting 'js' ...
[__:__:__] Starting 'js_myBundle' ...
  size  name  module                      status
  1 kB  0     ./js/script.js + 1 modules  built ✖
  1 kB  1     multi ./js/script.js        built

  (2 assets hidden)

  Failed to compile.

./js/Component.js

__PATH__/packages/integration/fixtures/crafty-preset-babel-webpack/lints/js/Component.js
  11:3   error  Opening curly brace does not appear on the same line as controlling statement  brace-style
  12:13  error  Multiple spaces found before '<'                                               no-multi-spaces

✖ 2 problems (2 errors, 0 warnings)
  2 errors, 0 warnings potentially fixable with the \`--fix\` option.

 @ ./js/script.js 7:0-39 23:33-42
 @ multi ./js/script.js

./js/script.js

__PATH__/packages/integration/fixtures/crafty-preset-babel-webpack/lints/js/script.js
   3:24  error    Multiple spaces found before '\\"./Component.js\\"'                                no-multi-spaces
   5:10  error    'test' is defined but never used                                               no-unused-vars
   5:19  error    A space is required after ','                                                  comma-spacing
   6:1   error    Expected indentation of 2 spaces but found 3                                   indent
  11:5   warning  Unexpected console statement                                                   no-console
  15:5   warning  Unexpected console statement                                                   no-console
  15:19  error    Multiple spaces found before '\\"some method\\"'                                   no-multi-spaces
  15:33  error    A space is required after ','                                                  comma-spacing
  15:33  error    There should be no space before ','                                            comma-spacing
  19:3   error    Opening curly brace does not appear on the same line as controlling statement  brace-style
  20:14  error    Expected '===' and instead saw '=='                                            eqeqeq

✖ 11 problems (9 errors, 2 warnings)
  7 errors, 0 warnings potentially fixable with the \`--fix\` option.

 @ multi ./js/script.js

  Δt ____ms ✖ 2 problems (2 errors, 0 warnings)

",
}
`;

exports[`Removes unused classes 1`] = `
Object {
  "status": 0,
  "stderr": "",
  "stdout": "
[__:__:__] Starting Crafty __version__...
[__:__:__] Files will be stored in:
            js: __PATH__/packages/integration/fixtures/crafty-preset-babel-webpack/tree-shaking/dist/js
[__:__:__] Starting 'default' ...
[__:__:__] Starting 'js' ...
[__:__:__] Starting 'js_myBundle' ...
  size  name     module                     status
  1 kB  0        ./js/index.js + 2 modules  built
  1 kB  1        multi ./js/index.js        built

  size  name     asset                      status
  1 kB  default  myBundle.min.js            emitted
  1 kB  default  myBundle.min.js.map        emitted



  Compiled successfully!
  Δt ____ms

[__:__:__] Finished 'js_myBundle' after ____ ms
[__:__:__] Finished 'js' after ____ ms
[__:__:__] Finished 'default' after ____ ms
",
}
`;
